Wrappers/Python/ccpi/viewer/ui/dialogs.py

```diff
@@ -137,6 +189,153 @@ def enableDisableDimZ(self):
             widgets['dim_Z_field'].setEnabled(True)
         else:
             widgets['dim_Z_field'].setEnabled(False)
+    
+    def preview(self):
+        pars = self.getRawAttrs()
+        
+        # retrieve info about image file from interface
+        dimensionality = [3, 2][self.getWidget('dimensionality').currentIndex()]
+        dimX, dimY, dimZ = pars['shape']
+        isFortran = pars['is_fortran']
+        isBigEndian = pars['is_big_endian']
+        # typecode = pars['typecode']
+        typecode = self.getWidget('dtype').currentText()
+
+        
+        if isFortran:
+            shape = (dimX, dimY)
+        else:
+            shape = (dimY, dimX)
+        if dimensionality == 3:
+            if isFortran:
+                shape = (dimX, dimY, dimZ)
+                central_slice = 'z'
+            else:
+                shape = (dimZ, dimY, dimX)
+                central_slice = 'x'
```

Comment:
I added
`if dimensionality == 3:
            if isFortran:
                shape = (dimX, dimY, dimZ)
                central_slice = 'z'
            else:
                shape = (dimZ, dimY, dimX)
                central_slice = 'x'`

This means that the axes in the preview are wrong. We need to rotate them.

Reply:
I did not test the code for dimensionality = 2, I need data for it.

Reply:
This is now added to the preview.

Reply:
Notice that we should also swap the axis labels when loading the whole image. This would be done in the application that uses the viewer, not in this dialog.